subprojects/docs/src/docs/userguide/running-builds/gradle_daemon.adoc

```diff
@@ -258,9 +258,53 @@ This message means that either
 --
 In both cases the build cannot benefit from file system watching.
 
-`Caught exception: Couldn't add watch, error = 28`::
-If you receive this error on Linux then you ran out of inotify handles.
-To raise the limit see https://confluence.jetbrains.com/display/IDEADEV/Inotify+Watches+Limit[here].
-
 `java.io.IOException: Too many open files`::
 If you receive this error on macOS, you need to raise your open files limit, see https://superuser.com/a/443168/8117[here].
+
+===  Linux specific notes
+
+File system watching uses http://en.wikipedia.org/wiki/Inotify[Inotify] on Linux.
+Depending on the size of your build, it may be necessary to increase Inotify limits.
+If you are using an IDE, then you probably already did increase the limits.
+
+[[sec:inotify_watches_limit]]
+Inotify watches limit::
+File system watching uses one Inotify watch per watched directory.
+You can see the current limit of Inotify watches per user by running:
++
+[source,bash]
+----
+cat /proc/sys/fs/inotify/max_user_watches
+----
++
+To increase the limit to e.g. 512K watches:
+
+1. Add the following line to either `/etc/sysctl.conf` or a new `*.conf` file under `/etc/sysctl.d/` directory:
++
+[source,properties]
+----
+fs.inotify.max_user_watches = 524288
+----
+2. Then run this command to apply the change:
++
+[source,bash]
+----
+sudo sysctl -p --system
+----
+
+[[sec:inotify_instances_limit]]
+Inotify instances limit::
```

Comment:
Do we need to talk about this case at all? I never saw anyone run into limitations with this one. If they do, we can just give the error message explaining that they ran out of inotify instances, and they can do their research from there.

Reply:
I removed the section about the instance limit. You are probably right that users don't seem to hit it in the wild.